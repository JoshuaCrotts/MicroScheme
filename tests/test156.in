;
; Another slightly simple OOP example.
;
(define (make-simple-cell value)
    (位 (selector)
        (cond ((eq? selector 'fetch)
               (位 () value))
              ((eq? selector 'store!)
               (位 (new-value)
                (set! value new-value)))
              ((eq? selector 'cell?)
               (位 () #t))
              (else "Not handled"))))

(define a-cell (make-simple-cell 13))
((a-cell 'fetch)) ;; 13
((a-cell 'store!) 21)
((a-cell 'cell?))
(define b-cell (make-simple-cell 17))
((b-cell 'fetch)) ;; 17
((b-cell 'store!) 24)
((b-cell 'cell?))
((a-cell 'fetch)) ;; 21
((b-cell 'fetch)) ;; 24
