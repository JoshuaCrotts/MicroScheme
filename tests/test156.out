13
#t
17
#t
21
24
